lib/srv/db/common/auth.go

```diff
@@ -1009,6 +1010,21 @@ func (a *dbAuth) getClientCert(ctx context.Context, expiry time.Time, databaseUs
 	return &clientCert, resp.CACerts, nil
 }
 
+// GenerateDatabaseClientKey generates a cryptographic key appropriate for
+// database client connections.
+func (a *dbAuth) GenerateDatabaseClientKey(ctx context.Context) (*keys.PrivateKey, error) {
+	signer, err := cryptosuites.GenerateKey(ctx,
+		cryptosuites.GetCurrentSuiteFromAuthPreference(a), cryptosuites.DatabaseClient)
```

Comment:
not fault of your change but we should probably switch to access point/cache for auth preference. i can do that separately.

Reply:
thanks - yeah currently this will add an additional GetAuthPreference for every key generated

one day we should probably start caching db client certificates too, the whole roundtrip to get auth to sign new certs for every call is probably still the bottleneck https://github.com/gravitational/teleport/blob/270a55b7c0f8d4041b84d87998f45d7ff9e6782b/lib/srv/db/common/auth.go#L993-L994